src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(toFileName, createReport(fromFileName));
+    }
 
+    private String createReport(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        String line;
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader inputFile = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = inputFile.readLine()) != null) {
+                String[] splittedLine = line.split(",");
+                if (splittedLine[0].equals("supply")) {
+                    supplyTotal += Integer.parseInt(splittedLine[1]);
+                } else if (splittedLine[0].equals("buy")) {
+                    buyTotal += Integer.parseInt(splittedLine[1]);
+                }
+            }
+            final int result = supplyTotal - buyTotal;
+            builder.append("supply,").append(supplyTotal).append(System.lineSeparator())
+              .append("buy,").append(buyTotal).append(System.lineSeparator())
+              .append("result,").append(result).append(System.lineSeparator());
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Cant read from file", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Cant read from file", e);
+        }
```

Comment:
Don't create two identical `catch` blocks. Either use `multiCatch` ([reference](https://jenkov.com/tutorials/java-exception-handling/catching-multiple-exceptions.html)).

But, here you have a hierarchy. `FileNotFound` is already an `IOException`. So remove the first `catch`.

<img width="574" alt="image" src="https://user-images.githubusercontent.com/68386737/235071501-f89e1992-0871-422e-ace5-66a999e27651.png">